src/Core/Containers/VectorArray.hpp

```diff
@@ -2,92 +2,93 @@
 
 #include <Core/Containers/AlignedStdVector.hpp>
 #include <Core/RaCore.hpp>
+#include <Core/Utils/ContainerIntrospectionInterface.hpp>
+
+#include <type_traits> // std::is_integral
 
 namespace Ra {
 namespace Core {
 
-/// This class is a wrapper around a std::vector of Core::Vectors.
-/// which allow to use the stdlib's dynamic array implementation, yet pass it as
-/// a matrix when Eigen needs it with the getMap() method.
+/// Internal class used to compute types and values used by VectorArray, to handle both
+/// eigen matrices and integral types
+template <typename V, bool isArithmetic, bool isEigen>
+struct VectorArrayTypeHelper {};
+
+/// This class is implements ContainerIntrospectionInterface for AlignedStdVector.
+/// It provides Eigen::Map functionality if the underlying component allows it (i.e. fixed size)
 template <typename V>
-class VectorArray : public AlignedStdVector<V>
+class VectorArray : public AlignedStdVector<V>, public Utils::ContainerIntrospectionInterface
 {
+  private:
+    using TypeHelper =
+        VectorArrayTypeHelper<V,
+                              std::is_arithmetic<V>::value || std::is_enum<V>::value,
```

Comment:
why not using `std::is_scalar<V>::value` that will also allow pointers as `component_type` ?